### PR TITLE
gdalinfo, ogrinfo: Improve error output when file cannotbe opened

### PR DESCRIPTION
**Pull request by dbaston**
## What does this PR do?

Updates `gdalinfo` to check if a file that cannot be opened has a name indicating a vector data type, and suggest the user call `ogrinfo` instead. Make the corresponding update to `ogrinfo`.

## What are related issues/pull requests?

#9618

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Review by elpaso — APPROVED** [review]

File: apps/gdalinfo_bin.cpp [hunk 1]
```diff
@@ -148,8 +148,32 @@ MAIN_START(argc, argv)
 #ifdef __AFL_HAVE_MANUAL_CONTROL
             continue;
 #else
-        fprintf(stderr, "gdalinfo failed - unable to open '%s'.\n",
-                psOptionsForBinary->pszFilename);
+        bool bMaybeVector = false;
+        VSIStatBuf sStat;
+        if (VSIStat(psOptionsForBinary->pszFilename, &sStat) == 0)
```
Comment (rouault) [inline apps/gdalinfo_bin.cpp:153]: what about using GDALIdentifyDriver() instead and test if it is a vector driver? That would be more robust as not all drivers are extension specific, and that would work with non-file connection name

Same remark for ogrinfo

Comment (dbaston) [reply apps/gdalinfo_bin.cpp:153]: Is there a potential cost to that? I worried about a bunch of filesystem or network activity just to generate a better error message, but that may be unfounded.

Comment (rouault) [reply apps/gdalinfo_bin.cpp:153]: That should be acceptable. Most network requests should already have bien cached. Abd this is an error code path so performance is less of a concern

Comment (dbaston) [reply apps/gdalinfo_bin.cpp:155]: I've made this change

**Pull request merged**